browser/scripts/account-controller.js

```diff
@@ -390,50 +401,76 @@ AccountController.prototype.openResetPasswordModal = function(dfd) {
 	return dfd.promise
 }
 
+AccountController.prototype.fillAccountForm = function() {
+	var formEl = $('#account-modal-form');
+	var nameIn = $('#name_id');
+	var unameIn = $('#username_id');
+	var emailIn = $('#email_id');
+	nameIn.val(E2.models.user.get('name'));
+	unameIn.val(E2.models.user.get('username'));
+	emailIn.val(E2.models.user.get('email'));
+	
+	if (nameIn.val())
+		nameIn.parent().find('label').addClass('filled-label');
+	if (unameIn.val())
+		unameIn.parent().find('label').addClass('filled-label');
+	if (emailIn.val())
+		emailIn.parent().find('label').addClass('filled-label');
+}
+
 AccountController.prototype.openAccountModal = function(dfd) {
 	var that = this;
 	var dfd = dfd || when.defer();
 	var accountTemplate = E2.views.account.account;
 	
-	ga('send', 'event', 'account', 'open', 'resetModal');
-	
+	//ga('send', 'event', 'account', 'open', 'accountModal');
+	console.trace();
 	var bb = bootbox.dialog(
 	{
 		show: true,
 		animate: false,
 		message: 'Rendering',
+		onEscape: function() {
+			dfd.resolve();
+		}
 	}).init(function() {
 		E2.app.useCustomBootboxTemplate(accountTemplate);
+		that.fillAccountForm();
 	});
 
-	this._bindEvents(bb, dfd);
+	//this._bindEvents(bb, dfd);
 	
 	var formEl = $('#account-modal-form');
-	formEl.submit(function( event )
-	{
+	formEl.submit(function( event ) {
 		event.preventDefault();
 		
 		var formData = formEl.serialize();
 
-		$.ajax(
-		{
+		$.ajax({
 			type: "POST",
 			url: formEl.attr('action'),
 			data: formData,
-			error: function(err)
-			{	
-				var errText = 'Account update failed.'
-				that.showError('general',errText);
+			error: function(err) {
+				if (err.responseJSON) {
+					err.responseJSON.map(function(ei) {
+						that.showError(ei.param, ei.msg)
+					})
+				} else {
+					var errText = 'Account update failed.'
+					that.showError('general', errText);
+				}
+			
+				$('#signup-form_id .required').parent().addClass('wrong');
 			},
 			success: function(user)
 			{
-				ga('send', 'event', 'account', 'accountUpdated', user.username)
+				ga('send', 'event', 'account', 'accountUpdated', user.username);
 				bootbox.hideAll();
-				bootbox.alert('Account updated!');
-				dfd.resolve();
+				window.location.reload();
```

Comment:
if this is just updating the account details then just update the user in environment
only when registering a new user, then reload the entire page. (for now)